nservicebus/databus.md

```diff
@@ -0,0 +1,69 @@
+---
+title: DataBus feature
+summary: DataBus feature
+tags:
+- DataBus
+- Attachments
+---
+
+## What is DataBus feature for?
+
+Messages are intended to be small. Some scenarios require to send large binary data along with a message. For this purpose, NServiceBus has a DataBus feature to allow to overcome limitations of message size.
+
+## How it works?
+
+`DataBus` is taking an approach of storing large payload and storing it in at a location both sending and receiving parties can access. Message is sent with a reference to the location, and upon processing payload is brought, allowing receiving part access message along with payload. In case location is not available upon sending, message will fail send operation. Receive operation will fail as well when payload location is not available and will cause standard NServiceBus behaviour, causing retries and eventually going into error queue.

```

Comment:
Apparently not always! Mauro is correct. Change it to "storing it in a
location".

On Mon, Jan 5, 2015 at 5:53 PM, Sean Feldman <notifications@github.com>
wrote:

> In nservicebus/databus.md
> <https://github.com/Particular/docs.particular.net/pull/534#discussion-diff-22468027>
> :
>
> > @@ -0,0 +1,69 @@
> > +---
> > +title: DataBus feature
> > +summary: DataBus feature
> > +tags:
> > +- DataBus
> > +- Attachments
> > +---
> > +
> > +## What is DataBus feature for?
> > +
> > +Messages are intended to be small. Some scenarios require to send large binary data along with a message. For this purpose, NServiceBus has a DataBus feature to allow to overcome limitations of message size.
> > +
> > +## How it works?
> > +
> > +`DataBus` is taking an approach of storing large payload and storing it in at a location both sending and receiving parties can access. Message is sent with a reference to the location, and upon processing payload is brought, allowing receiving part access message along with payload. In case location is not available upon sending, message will fail send operation. Receive operation will fail as well when payload location is not available and will cause standard NServiceBus behaviour, causing retries and eventually going into error queue.
>
> @AnnePicus <https://github.com/AnnePicus> is a spelling magician
> (spellician) :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Particular/docs.particular.net/pull/534/files#r22468027>
> .
>



-- 
Anne